### PR TITLE
[staging] tracker: 3.0.3 -> 3.1.1

### LINKED ISSUE #118155: Steam / gtk+3 broken on master
`gtk+3` is broken on master

```
       > FAILED: docs/reference/libtracker-sparql/examples/writeonly-example.p/writeonly-example.c.o
       > gcc -Idocs/reference/libtracker-sparql/examples/writeonly-example.p -Idocs/reference/libtracker-sparql/examples -I../docs/reference/libtracker-sparql/examples -Isrc -I../src -I/nix/store/w98bzaw8cgpd3h9vwhg0qs2qgm96wxg0-glib-2.66.8-dev/include/glib-2.0 -I/nix/store/l8k9ji93sb9yd6mfig1jgh4g0imr7ygx-glib-2.66.8/lib/glib-2.0/include -I/nix/store/w98bzaw8cgpd3h9vwhg0qs2qgm96wxg0-glib-2.66.8-dev/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wformat -Wformat-security -Wno-pointer-sign -DTRACKER_COMPILATION '-DG_LOG_DOMAIN="Tracker"' -DG_LOG_STRUCTURED=1 -MD -MQ docs/reference/libtracker-sparql/examples/writeonly-example.p/writeonly-example.c.o -MF docs/reference/libtracker-sparql/examples/writeonly-example.p/writeonly-example.c.o.d -o docs/reference/libtracker-sparql/examples/writeonly-example.p/writeonly-example.c.o -c ../docs/reference/libtracker-sparql/examples/writeonly-example.c
       > In file included from ../docs/reference/libtracker-sparql/examples/writeonly-example.c:1:
       > ../src/libtracker-sparql/tracker-sparql.h:36:10: fatal error: libtracker-sparql/tracker-sparql-enum-types.h: No such file or directory
       >    36 | #include <libtracker-sparql/tracker-sparql-enum-types.h>
       >       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > compilation terminated.
       > [32/168] Generating tracker-enum-types.h with a custom command (wrapped by meson to capture output)ls[Kresdump.vala src/libtracker-sparql/tracker-sparql-3.0.vapi
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/rpaqyf426i2lv6q478kzfy2882cxgp30-tracker-3.0.3.drv'.
 ```
 The header is from `gnome3.tracker`, which seems to be passed
 ```
 $ nix-locate libtracker-sparql/tracker-sparql-enum-types.h
gnome3.tracker.dev                                  916 r /nix/store/cqdzv09xdwqbkz9ymajgmch5fs5n4lvb-tracker-3.0.1-dev/include/tracker-3.0/libtracker-sparql/tracker-sparql-enum-types.h
$ nix show-derivation /nix/store/biqlzpi9blxhqa270ppckdlz42visxbr-gtk+3-3.24.27.drv | grep tracker
      "/nix/store/rpaqyf426i2lv6q478kzfy2882cxgp30-tracker-3.0.3.drv": [
      "buildInputs": "/nix/store/ldzagpm90b9577ibxk6k73sgk8knm64k-libxkbcommon-1.1.0-dev /nix/store/3w4hvwb5bjrhafx5g0ahj020p8vy68zk-epoxy-1.5.4-dev /nix/store/s66rfjj9rpwbi2dls044jjn527x8vxd2-json-glib-1.6.2-dev /nix/store/z526yrbi0lmpcfcam8cccqgpmpj4h34d-iso-codes-4.6.0 /nix/store/w7glqhm22zhscawjhy1nv382ypr3d8cc-tracker-3.0.3-dev",
      "mesonFlags": "-Dgtk_doc=true -Dtests=false -Dtracker3=true -Dbroadway_backend=true",
```
I haven't pin-pointed the failing commit yet, but it looks like meson isn't getting the include path correctly.

cc @NixOS/gnome

### PR DESCRIPTION
**Pull request by bkchr**
###### Motivation for this change

Fixes: https://github.com/NixOS/nixpkgs/issues/118155

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118365` at 0b954102 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>504 packages marked as broken and skipped:</summary>
  <ul>
    <li>adoptopenjdk-hotspot-bin-13</li>
    <li>adoptopenjdk-hotspot-bin-14</li>
    <li>adoptopenjdk-jre-hotspot-bin-13</li>
    <li>adoptopenjdk-jre-hotspot-bin-14</li>
    <li>agdaPackages.iowa-stdlib</li>
    <li>aldor</li>
    <li>apostrophe</li>
    <li>aqemu</li>
    <li>asymptote</li>
    <li>belle-sip</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/19c4829a379afa1c8e542e7b33f2f0f0">gupnp-igd</a></li>
    <li><a href="https://gist.github.com/05b830c01b720e227e9c63595e42e721">libnice</a></li>
  </ul>
</details>
<details>
  <summary>2906 packages skipped due to time constraints:</summary>
  <ul>
    <li>AusweisApp2</li>
    <li>DisnixWebService</li>
    <li>MIDIVisualizer</li>
    <li>R</li>
    <li>abcl</li>
    <li>abiword</li>
    <li>adapta-gtk-theme</li>
    <li>adementary-theme</li>
    <li>adoptopenjdk-hotspot-bin-15</li>
    <li>adoptopenjdk-hotspot-bin-16</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>28 packages built successfully:</summary>
  <ul>
    <li>adoptopenjdk-bin (adoptopenjdk-hotspot-bin-11 ,openjdk11-bootstrap)</li>
    <li>openjdk8-bootstrap (adoptopenjdk-hotspot-bin-8)</li>
    <li>appgate-sdp</li>
    <li>czkawka</li>
    <li>geoclue2</li>
    <li>glib-networking (gnome3.glib-networking ,gnome3.glib_networking)</li>
    <li>gnome3.gnome-desktop (gnome3.gnome_desktop)</li>
    <li>gnome3.gnome-session-ctl</li>
    <li>gtk3 (gnome3.gtk ,gtk3-x11)</li>
    <li>tracker (gnome3.tracker)</li>
    <li>gssdp</li>
    <li>gtkspell3</li>
    <li>gupnp</li>
    <li>libcanberra-gtk3</li>
    <li>libdbusmenu-gtk3</li>
    <li>libproxy</li>
    <li>libwnck3</li>
    <li>mpdevil</li>
    <li>networkmanager</li>
    <li>nice-dcv-client</li>
    <li>openconnect (openconnect_gnutls)</li>
    <li>pantheon.elementary-icon-theme</li>
    <li>pantheon.granite</li>
    <li>prs</li>
    <li>stoken</li>
    <li>usbview</li>
    <li>vscode</li>
    <li>xfce.xfce4-dev-tools (xfce.xfce4_dev_tools)</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

File: pkgs/development/libraries/tracker/default.nix [hunk 1]
```diff
@@ -29,20 +29,21 @@
...
-    sha256 = "sha256-b1yEqzvh7aUgUBsq7XIhYWoM8VKRDFN3V7U4vAXv/KM=";
+    sha256 = "sha256:0x98wk9347p9sf7vw0yhyswnzmd8hp01zczl5a0fyiw7fnaynmfa";
   };
 
   patches = [
     (substituteAll {
       src = ./fix-paths.patch;
       inherit asciidoc;
     })
+    ./fix-docs.patch
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/tracker/default.nix:46]: Is there an upstream patch we can fetch instead?

**Review by SuperSandro2000 — COMMENTED** [review]
/rebase staging

**Comment by rmcgibbo** [comment]
> 3 packages failed to build:

The bot failures look unrelated to this PR. Sorry for the noise.

File: pkgs/development/libraries/tracker/default.nix (hunk 1 above)
Comment (bkchr) [reply pkgs/development/libraries/tracker/default.nix:46]: Not yet, I have created this patch on my own.

**Comment by SFrijters** [comment]
I tried this PR to fix my steam issues - it gives a different error for me than before, but an error nonetheness:

```
$ $ NIXPKGS_ALLOW_UNFREE=1 nix-build -K -A steam
[...]
/nix/store/8s4qrkjjas2bk3i91il1jkcan094in60-gcc-10.2.0/lib/gcc/i686-unknown-linux-gnu/10.2.0/include/stddef.h:424: syntax error, unexpected identifier in '  __float128 __max_align_f128 __attribute__((__aligned__(__alignof(__float128))));' at '__float128'
[...]
▶ 25/32 Tracker-FATAL-ERROR: --- /build/tracker-3.1.0/tests/libtracker-data/datetime/direct-1.out       2021-03-21 18:01:57.491018000 +0000 ERROR         
25/32 tracker:data+slow / sparql                              ERROR            1.36s   killed by signal 5 SIGTRAP
>>> TRACKER_LANGUAGE_STOP_WORDS_DIR=/build/tracker-3.1.0/src/libtracker-common/stop-words LANG=en_US.UTF8 MALLOC_PERTURB_=212 /build/tracker-3.1.0/build/tests/libtracker-data/tracker-sparql-test
 ✀  
stderr:

(/build/tracker-3.1.0/build/tests/libtracker-data/tracker-sparql-test:2144): Tracker-ERROR **: 08:08:32.104: --- /build/tracker-3.1.0/tests/libtracker-data/datetime/direct-1.out       2021-03-21 18:01:57.491018000 +0000
+++ -   2021-04-03 08:08:32.103820215 +0000
@@ -1 +1 @@
-"0000-01-01T00:00:00Z" "2020-01-01T00:00:00Z"  "9999-01-01T00:00:00Z"
+"1969-12-31T23:59:59Z" "2020-01-01T00:00:00Z"  "1969-12-31T23:59:59Z"



TAP parsing error: unexpected input at line (253,)
(test program exited with status code -5)

```

Tracker gets built two times, looks like the second one is 32bit?

**Comment by SFrijters** [comment]
Note that tracker-3.1.1 has been released a couple of hours ago (I'd hope this would've fixed it), feel free to take commits from https://github.com/SFrijters/nixpkgs/tree/steam-fix (adds new `tappy` package needed for testing).

**Comment by bkchr** [comment]
Ty @SFrijters :)

**Comment by SFrijters** [comment]
I'm still seeing the failing tests, but I'm not sure if its a me problem, a we (nixpkgs) problem or a they (upstream) problem.
I don't advocate using it, but https://github.com/SFrijters/nixpkgs/commit/8aca7cc75eb37a1f77026bfd1fc126cae4ec804e 'fixes' it for me locally.

**Comment by jonringer** [comment]
since this change causes over 500 rebuilds, I moved it to target staging

See https://nixos.org/nixpkgs/manual/#submitting-changes-staging-branch for more details on staging branch

**Comment by jonringer** [comment]
rebased failed... Please do the following:

```
git rebase --onto=staging HEAD~2
git push .. ... --force
```
then change the base branch in the github PR from `master` -> `staging`

**Comment by bkchr** [comment]
> rebased failed... Please do the following:
> 
> ```
> git rebase --onto=origin/staging HEAD~2
> git push .. ... --force
> ```
> 
> then change the base branch in the github PR from `master` -> `staging`

Done

**Comment by avnik** [comment]
Folks, feel free to use stuff from #118439.  I tried to make own solution independently (just because my master rebuild fails, and I not found this one earlier).

File: pkgs/development/python-modules/tappy/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  version = "3.0";
+
+  disabled = pythonOlder "3.5";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-9e7u6/1k5T0yZhdSu0wohYmjuru5bbPzkaTsKfE1nHA=";
+  };
+
+  checkInputs = [ pytestCheckHook ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/tappy/default.nix:19]: Missing pythonImportsCheck.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
@ofborg eval

**Comment by avnik** [comment]
I understand why it going staging -- due amount of changes, but also I feel master won't able build w/o it

**Comment by SuperSandro2000** [comment]
> I understand why it going staging -- due amount of changes, but also I feel master won't able build w/o it

We can't merge 5000+ rebuilds into master.

**Comment by avnik** [comment]
> > I understand why it going staging -- due amount of changes, but also I feel master won't able build w/o it
> 
> We can't merge 5000+ rebuilds into master.

5k not-builds I believe.

File: pkgs/development/python-modules/tappy/default.nix (hunk 2 above)
Comment (bkchr) [reply pkgs/development/python-modules/tappy/default.nix:19]: Done

**Comment by vcunat** [comment]
> 5k not-builds I believe.

Not really, for 64 bits we do have binaries on Hydra.

**Comment by SFrijters** [comment]
I've been doing some digging in the code and I think the failing datetime test I'm seeing is an upstream thing (and I've created an issue for it https://gitlab.gnome.org/GNOME/tracker/-/issues/292). So maybe it's best to indeed disable the two tests that I patched out for now, but perhaps there is a nicer way to tell Meson to skip the tests?

**Comment by lovesegfault** [comment]
We might want to pull this patch from upstream that (allegedly) solves the issue: https://gitlab.gnome.org/GNOME/tracker/-/commit/57386f7e5a722a4b24ec503cb4cf1da57b44588c

**Comment by SFrijters** [comment]
I just built using https://gitlab.gnome.org/GNOME/tracker/-/merge_requests/401 and it fixes the issue(s) for me. The developer is actively responsing to my issue so I guess we will not have to wait long to at least take a commit from the master branch, or we can ask if there can be an actual release for it.

**Comment by SFrijters** [comment]
Copy-pasting my reply from the issue thread for increased visibility:

Looking at the commits we need more than one patch since e.g. dad7a4b6 in the same MR adds the helper functions that are used to fix the issue. Quoting upstream https://gitlab.gnome.org/GNOME/tracker/-/issues/292#note_1076661

> GNOME 40.1 is scheduled for May 1st from a look at https://wiki.gnome.org/Forty. I'll be releasing Tracker 3.1.2 around then the latest :). In the mean time it's fair to pick the fix.

I don't know if there is a nice nix(pkgs) way to pick a range of commits as patches? Alternatively, is it allowed to go to version `unstable-2021-04-07` and use the non-master commit e1c060e3e495491ee424d21cb2d10abcd474f793 for the time being?

**Comment by vcunat** [comment]
I often utilize `fetchpatch` with capabilities of GitLab/GitHub, e.g. https://gitlab.gnome.org/GNOME/tracker/-/merge_requests/401.diff

**Comment by SFrijters** [comment]
Ah, that seems the way to go then in this case. @bkchr Can you add that and strip off my WIP commit that disables the tests?

**Comment by bkchr** [comment]
Done

File: pkgs/development/libraries/tracker/default.nix [hunk 3]
```diff
@@ -29,20 +30,27 @@
...
+    # Fix 32bit datetime issue, use this upstream patch until 3.1.2 lands
+    # https://gitlab.gnome.org/GNOME/tracker/-/issues/292
+    (fetchpatch {
+      url = "https://gitlab.gnome.org/GNOME/tracker/-/merge_requests/401.diff";
```
Comment (jtojnar) [inline pkgs/development/libraries/tracker/default.nix:51]: Note that this patch might change if someone pushes to the branch before it is merged.

Comment (bkchr) [reply pkgs/development/libraries/tracker/default.nix:51]: So I assume it is better to  include the patch in nixpkgs directly for now?

Comment (vojta001) [reply pkgs/development/libraries/tracker/default.nix:51]: Or fetch directly by commit hash

**Comment by vojta001** [comment]
Not a deal breaker, but currently ebf35ae8531b86fbc1313e00f2c388b04b4be0ca goes before 7403c6c378ea80c4f76c3b24e60f4fa1337ca528 despite tappy being used in the tracker update. Probably a result of history rewrites

File: pkgs/development/libraries/tracker/default.nix (hunk 3 above)
Comment (bkchr) [reply pkgs/development/libraries/tracker/default.nix]: I added it now.

**Comment by bkchr** [comment]
> Not a deal breaker, but currently [ebf35ae](https://github.com/NixOS/nixpkgs/commit/ebf35ae8531b86fbc1313e00f2c388b04b4be0ca) goes before [7403c6c](https://github.com/NixOS/nixpkgs/commit/7403c6c378ea80c4f76c3b24e60f4fa1337ca528) despite tappy being used in the tracker update. Probably a result of history rewrites

Also fixed.

File: pkgs/development/libraries/tracker/default.nix [hunk 4]
```diff
@@ -76,6 +81,7 @@ stdenv.mkDerivation rec {
 
   checkInputs = [
     python3.pkgs.pygobject3
+    python3.pkgs.tappy
   ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/tracker/default.nix:85]:
```suggestion
  checkInputs = (with python3.pkgs; [
    pygobject3
    tappy
  ]);
```

**Comment by jtojnar** [comment]
Cherry-picked to #117081

**Comment by bkchr** [comment]
So this can be closed?

**Comment by jtojnar** [comment]
Yes. Thanks.

File: pkgs/development/libraries/tracker/default.nix (hunk 1 above)
Comment (jtojnar) [reply pkgs/development/libraries/tracker/default.nix:47]: Any progress in sending this upstream?

Comment (SFrijters) [reply pkgs/development/libraries/tracker/default.nix:47]: @jtojnar Upstream will merge this patch for the 3.1.2 release, probably in the next two weeks-ish, see https://gitlab.gnome.org/GNOME/tracker/-/issues/292#note_1076661 .

**Pull request closed without merging**